compiler/ilgen/IlBuilder.cpp

```diff
@@ -1380,24 +1392,59 @@ IlBuilder::EqualTo(TR::IlValue *left, TR::IlValue *right)
    return returnValue;
    }
 
+void
+IlBuilder::integerizeAddresses(TR::IlValue **leftPtr, TR::IlValue **rightPtr)
+   {
+   TR::IlValue *left = *leftPtr;
+   TR::IlValue *right = *rightPtr;
+   if (left->getSymbol()->getDataType() == TR::Address)
+      {
+      TR_ASSERT(right->getSymbol()->getDataType() == TR::Address, "mismatched types in address compare");
```

Comment:
I see how this function would mostly be useful when comparing address, but assuming that this function will only ever be called for comparisons doesn't seem quite right. Maybe the assert message should say something like "both types are not addresses", just to make it clear that it failed in this function as opposed to the actual comparison function.